cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -312,43 +505,226 @@ def migrate_from_modulestore(
+            source_data = source_data_list[i]
+
+            # Start migration for `source_pk`
+            # Staging legacy blocks
+            status.set_state(f"{SUB_STEP_MIGRATING} ({source_pk}): {MigrationStep.STAGING.value}")
+            staged_content = staging_api.stage_xblock_temporarily(
+                block=legacy_root_list[i],
+                user_id=status.user.pk,
+                purpose=CONTENT_STAGING_PURPOSE_TEMPLATE.format(source_key=source_data.source.key),
             )
-        else:
-            log.warning("No target entities found to add to collection")
+            migrations[i].staged_content = staged_content
+            status.increment_completed_steps()
+
+            # Parsing OLX
+            status.set_state(f"{SUB_STEP_MIGRATING} ({source_pk}): {MigrationStep.PARSING.value}")
+            parser = etree.XMLParser(strip_cdata=False)
+            try:
+                root_node = etree.fromstring(staged_content.olx, parser=parser)
+            except etree.ParseError as exc:
+                status.fail(f"Failed to parse source OLX (from staged content with id = {staged_content.id}): {exc}")
+            status.increment_completed_steps()
+
+            # Importing assets
+            status.set_state(f"{SUB_STEP_MIGRATING} ({source_pk}): {MigrationStep.IMPORTING_ASSETS.value}")
+            content_by_filename = _import_assets(migrations[i])
+            status.increment_completed_steps()
+
+            # Importing structure of the legacy block
+            status.set_state(
+                f"{SUB_STEP_MIGRATING} ({source_pk}): {MigrationStep.IMPORTING_STRUCTURE.value}"
+            )
+            change_log, root_migrated_node = _import_structure(
+                migrations[i],
+                source_data,
+                target_library,
+                content_by_filename,
+                root_node,
+                status,
+            )
+            migrations[i].change_log = change_log
+            status.increment_completed_steps()
+
+            status.set_state(f"{SUB_STEP_MIGRATING} ({source_pk}): {MigrationStep.UNSTAGING.value}")
+            staged_content.delete()
+            status.increment_completed_steps()
+
+            _create_migration_artifacts_incrementally(
+                root_migrated_node=root_migrated_node,
+                source=source_data.source,
+                migration=migrations[i],
+                status=status,
+                source_pk=source_pk,
+            )
+            status.increment_completed_steps()
+
+    # Forwarding legacy content to migrated content
+    status.set_state(MigrationStep.FORWARDING.value)
+    if forward_source_to_target:
+        for i in range(len(migrations)):
+            _forwarding_content(migrations[i], source_data_list[i])
+    status.increment_completed_steps()
+
+    # Populating collections
+    status.set_state(MigrationStep.POPULATING_COLLECTION.value)
+    from .api import get_migration_info
```

Comment:
Seems odd to have this import in the middle of the function. Please add a comment explaining why (if it's necessary).

Reply:
Updated https://github.com/openedx/edx-platform/pull/37381/commits/6b653d4005a7612e49dbfe16310d80828eec2fc9